unison-core/src/Unison/Name.hs

```diff
@@ -128,16 +149,18 @@ parent (Name txt) = case unsnoc (Text.splitOn "." txt) of
   Nothing -> Nothing
   Just ([],_) -> Nothing
   Just (init,_) -> Just $ Name (Text.intercalate "." init)
-parent (Name' names) =
-  Name' <$> NonEmpty.nonEmpty (NonEmpty.init names)
+parent (Name' p names) =
+  Name' p <$> NonEmpty.nonEmpty (NonEmpty.init names)
 
 suffixes :: Name -> [Name]
 suffixes (Name n) =
   fmap up . tails . dropWhile (== "") $ Text.splitOn "." n
   where
   up ns = Name (Text.intercalate "." ns)
-suffixes (Name' names) =
-  map (Name' . NonEmpty.fromList) (NonEmpty.init (NonEmpty.tails names))
+suffixes name@(Name' p names) =
+  case p of
+    Absolute -> error ("suffixes (" ++ show name ++ ")")
+    Relative -> map (Name' p . NonEmpty.fromList) (NonEmpty.init (NonEmpty.tails names))
```

Comment:
Probably `Absolute` is okay here?  e.g. if the Name `.foo.bar.Baz` exists, suffix-based resolution should find it with as `Baz` or as `bar.Baz`.  Worth noting that the result of this suffixes operation is all `Relative` names.

Reply:
Can probably just ignore `p`.

Reply:
Ok, so you mean this?
```haskell
suffixes .foo.bar.baz = [foo.bar.baz, bar.baz, baz]
```
I like that. I wasn't sure if that first entry should be absolute or relative, so I put an error.

Reply:
Yeah; I think that's what the current stringy implementation does, right?  `dropWhile (== "")` effectively making everything relative first.